resolver/resolver.cc

```diff
@@ -3643,6 +3695,40 @@ class ResolveSignaturesWalk {
             [&](const ast::ExpressionPtr &e) {});
     }
 
+    static bool hasCompatibleOverloadedSigsWithKwArgs(core::Context ctx, int numSigs, const SigInformation &info,
+                                                      const std::vector<ArgInformation> &args) {
+        if (numSigs != 2) {
+            return false;
+        }
+        if (!info.allArgsMatched) {
+            return false;
+        }
+        if (info.hasMissingArgument) {
+            return false;
+        }
+
+        const auto &argv0 = args[0];
+        const auto &argv1 = args[1];
+
+        // Unlike std::equal, absl::c_equal will test for equal sizes.
+        auto argsEqual = [&ctx](const auto &arg0, const auto &arg1) {
+            if (arg0.name != arg1.name) {
+                return false;
+            }
+            return core::Types::equiv(ctx, arg0.type, arg1.type);
+        };
+
+        // TODO(froydnj) better error messages for users trying to provide overloads with kwargs?
```

Comment:
What would the error message look like right now?

With more time what would you have it show instead?

Reply:
The error message right now is just "Overloaded functions cannot have keyword arguments", pointing at all relevant keyword arguments.  It is basically what you get now if you try to define overloaded methods with keyword arguments, but it's not entirely accurate in this case, because we do allow it, we just have a lot of restrictions on it.

At this point in the code, we know that the user is probably trying to define an overloaded method with keyword arguments; we could issue more specific messages for each case from here on out:

* Mismatched positional and/or keyword argument names;
* Mismatched positional and/or keyword argument types;
* Both or neither sigs specifying a type for any associated block argument.